server/src/joinhandler.rs

```diff
@@ -167,8 +173,8 @@ impl<'a> System<'a> for JoinHandlerSystem {
                         continue;
                     }
 
-                    let spawn_block_pos =
-                        BlockPosition::new(level.spawn_x, level.spawn_y, level.spawn_z);
+                    let pos = positions.get(player).unwrap();
+                    let spawn_block_pos = pos.current.block_pos();
 
                     let spawn_position = SpawnPosition::new(spawn_block_pos);
```

Comment:
The `SpawnPosition` packet is used to send the world's spawn position, i.e. the location where a compass would point to. This change should be reverted.

Reply:
Ah, right. The PlayerPositionAndLookClientbound packet below will need to use different values then.